File: sme-creditmind-ai-backend/app/schemas/credit_score.py
from __future__ import annotations

from typing import Optional, Literal, List

from pydantic import BaseModel


class CreditFactorOut(BaseModel):
    name: str
    impact: Literal["positive", "negative", "neutral"]
    weight: float
    description: str
    dataPoint: str
    category: Literal["revenue", "volume", "customers", "consistency", "digital", "seasonal", "growth"]


class CreditScoreOut(BaseModel):
    id: str
    merchantId: str
    score: int
    baseScore: int
    aiAdjustment: int
    grade: Literal["A", "B", "C", "D", "E"]
    riskLevel: Literal["VERY_LOW", "LOW", "MEDIUM", "HIGH", "VERY_HIGH"]
    preApprovedLimit: float
    confidence: float
    factors: List[CreditFactorOut]
    narrative: str
    recommendation: Literal["APPROVE", "REVIEW", "DECLINE"]
    assessedBy: Optional[str] = None
    validUntil: str
    createdAt: str

File: sme-creditmind-ai/src/types/credit-score.ts
import type { CreditGrade, RiskLevel, Recommendation } from "./merchant";

export interface CreditScore {
  id: string;
  merchantId: string;
  score: number;
  baseScore: number;
  aiAdjustment: number;
  grade: CreditGrade;
  riskLevel: RiskLevel;
  preApprovedLimit: number;
  confidence: number;
  factors: CreditFactor[];
  narrative: string;
  recommendation: Recommendation;
  assessedBy?: string;
  validUntil: string;
  createdAt: string;
}

export interface CreditFactor {
  name: string;
  impact: "positive" | "negative" | "neutral";
  weight: number;
  description: string;
  dataPoint: string;
  category: "revenue" | "volume" | "customers" | "consistency" | "digital" | "seasonal" | "growth";
}

export interface ScoreHistory {
  date: string;
  score: number;
  grade: CreditGrade;
}
